src/components/Modal/index.tsx

```diff
@@ -5,13 +5,13 @@ import * as StyleUtils from '@styles/StyleUtils';
 import themeColors from '@styles/themes/default';
 import CONST from '@src/CONST';
 import BaseModal from './BaseModal';
-import {defaultProps, propTypes} from './modalPropTypes';
+import BaseModalProps from './types';
 
-function Modal(props) {
-    const [previousStatusBarColor, setPreviousStatusBarColor] = useState();
+function Modal({fullscreen, onModalHide, type, onModalShow, children, ...rest}: BaseModalProps) {
```

Comment:
Please define default values same as the ones in defaultProps before migration.
i.e. `onModalHide()` will crash in `hideModal` function

Reply:
Good example of regression because of `fullscreen` not defaulting to true:

Before:

![Simulator Screenshot - iPhone 15](https://github.com/Expensify/App/assets/108292595/c0bd6658-c92a-4e26-9217-955ac81afd5c)

After:

![Simulator Screenshot - iPhone 15](https://github.com/Expensify/App/assets/108292595/9ab79db4-2f47-42c0-8dc6-44218e7411e6)